sdk/keyvault/keyvault-keys/src/index.ts

```diff
@@ -419,6 +420,31 @@ export class KeyClient {
     });
   }
 
+  /**
+   * Gets a {@link CryptographyClient} for the given key.
+   *
+   * Example usage:
+   * ```ts
+   * let client = new KeyClient(url, credentials);
+   * // get a cryptography client for a given key
+   * let cryptographyClient = client.getCryptographyClient("MyKey");
+   * ```
+   * @param name - The name of the key used to perform cryptographic operations.
+   * @param version - Optional version of the key used to perform cryptographic operations.
+   * @returns - A {@link CryptographyClient} using the same options, credentials, and http client as this {@link KeyClient}
+   */
+  public getCryptographyClient(name: string, version?: string): CryptographyClient {
+    // The goals of this method are discoverability and performance (by sharing a client and pipeline).
+    // The existing cryptography client does not accept a pipeline as an argument, nor does it expose it
+    // so we _fake_ it by constructing it with the keyUrl, a fake token (which will be thrown away), and then
+    // set the remote provider's generated client to our current client.
+    const keyUrl = new URL(["keys", name, version].filter(Boolean).join("/"), this.vaultUrl);
```

Comment:
I like this!

Reply:
it's a little terser than I usually write (I think code should be boring) but I like that it handles version being falsy as well as any trailing slashes cleanly. Thanks!